### PR TITLE
Detect Classic SignalR Server

### PR DESCRIPTION
**Pull request by mikaelm12**
Fixes: https://github.com/aspnet/AspNetCore/issues/7781
Detect a user trying to connect to a classic SignalR Server with a Core client and guide them in the right direction.
Related: https://github.com/aspnet/AspNetCore/issues/7705

### CODE REVIEW

File: src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java [hunk 1]
```diff
@@ -28,6 +28,10 @@ public NegotiateResponse(String negotiatePayload) {
                     case "error":
                         this.error = reader.nextString();
                         break;
+                    case "ProtocolVersion":
+                        this.error = "Detected a connection attempt to an ASP.NET SignalR Server. This client only supports connecting to an ASP.NET Core SignalR Server. See https://aka.ms/signalr-core-differences for details.";
+                        reader.skipValue();
```
Comment (BrennanConroy) [inline src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java]: should we just return immediately here?

File: src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java [hunk 2]
```diff
@@ -28,6 +28,10 @@ public NegotiateResponse(String negotiatePayload) {
                     case "error":
                         this.error = reader.nextString();
                         break;
+                    case "ProtocolVersion":
+                        this.error = "Detected a connection attempt to an ASP.NET SignalR Server. This client only supports connecting to an ASP.NET Core SignalR Server. See https://aka.ms/signalr-core-differences for details.";
```
Comment (anurse) [inline src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java]: "Detected a connection attempt" sounds a little awkward and I think we can simplify a bit.

```suggestion
                        this.error = "Detected an ASP.NET SignalR Server. This client only supports connecting to an ASP.NET Core SignalR Server. See https://aka.ms/signalr-core-differences for details.";
```

**Review by analogrelay — APPROVED** [review]
👍

File: src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java (hunk 1 above)
Comment (anurse) [reply src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java]: That would be reasonable. There's no need to finish parsing, the rest is garbage anyway.

Comment (mikaelm12) [reply src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/NegotiateResponse.java]: 👍

**Pull request merged**